src/components/TeamMemberTasks/TeamMemberTasks.jsx

```diff
@@ -81,47 +88,33 @@ const TeamMemberTasks = props => {
   }, [currentUserId]);
 
   useEffect(() => {
-    renderTeamsList();
-  }, []);
-
-  useEffect(() => {
-    submitTasks();
-    if (userId !== props.auth.user.userid) {
-      dispatch(fetchTeamMembersTask(userId, props.auth.user.userid));
-      const currentUserRole = getUserRole(userId)
-        .then(resp => resp)
-        .then(user => {
-          setUserRole(user.data.role);
-        });
-    } else {
-      dispatch(fetchTeamMembersTask(userId, null));
-      setUserRole(props.auth.user.role);
+    if (isLoading === false && shouldRun) {
+      renderTeamsList();
+      closeMarkAsDone();
     }
-  }, [updatedTasks]);
+  }, [usersWithTasks, shouldRun]);
 
   const closeMarkAsDone = () => {
+    setClickedToShowModal(false);
     setMarkAsDoneModal(false);
+    setCurrentUserId('');
   };
 
   const onUpdateTask = (taskId, updatedTask) => {
     const newTask = {
       updatedTask,
       taskId,
     };
-    setTasks(tasks => {
-      const tasksWithoutTheUpdated = [...tasks];
-      const taskIndex = tasks.findIndex(task => task._id === taskId);
-      tasksWithoutTheUpdated[taskIndex] = updatedTask;
-      return tasksWithoutTheUpdated;
-    });
-    setUpdatedTasks(tasks => [...tasks, newTask]);
+    submitTasks(newTask);
+    dispatch(fetchTeamMembersTask(userId, props.auth.user.userid, false));
   };
 
-  const submitTasks = async () => {
-    for (let i = 0; i < updatedTasks.length; i += 1) {
-      const updatedTask = updatedTasks[i];
-      const url = ENDPOINTS.TASK_UPDATE(updatedTask.taskId);
-      axios.put(url, updatedTask.updatedTask).catch(err => console.log(err));
+  const submitTasks = async updatedTasks => {
+    const url = ENDPOINTS.TASK_UPDATE(updatedTasks.taskId);
+    try {
+      await axios.put(url, updatedTasks.updatedTask);
+    } catch (error) {
+      console.log(error);
```

Comment:
Does the app alert the user of any failed calls anywhere else? Wondering if this should be displayed to the user in some notification or toast.

Reply:
Out of the scope of this PR.